### PR TITLE
Fix registration after an unsafe one

### LINKED ISSUE #286: `register-unsafely-without-email` config key is kept when turning `unsafe_registration` back to `false`
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 6/7
- Distribution: Tested on Debian, all affected
- Module version: 8.0.2

## How to reproduce (e.g Puppet code you use)

First, apply a simple catalog using `unsafe_registration` sets to `true`
```puppet
class { 'letsencrypt':
  unsafe_registration => true,
}
```

Next apply a catalog using `unsafe_registration` sets to `false`
```puppet
class { 'letsencrypt':
  email => 'hello@example.com',
}
```

## What are you seeing

`register-unsafely-without-email` is set to `true` in config file.

## What behaviour did you expect instead

No `register-unsafely-without-email` in config file, or set to `false`.

## Any additional information you'd like to impart

PR is coming…

### PR DESCRIPTION
**Pull request by neomilium**
This PR allow user to register safely after using `unsafe_registration` flag.

Issue details at #286 

#### This Pull Request (PR) fixes the following issues

Fixes #286

### CODE REVIEW

**Comment by ekohl** [comment]
Tests look red :(

**Comment by neomilium** [comment]
Fixed, thanks @ekohl

File: manifests/config.pp [hunk 1]
```diff
@@ -29,17 +29,23 @@
     $_config = $config
   }
 
+  $ensure_unsafe_registration = $unsafe_registration ? {
+    true  => present,
+    false => absent,
+  }
+
+  ini_setting { "${config_file} register-unsafely-without-email true":
+    ensure  => $ensure_unsafe_registration,
```
Comment (ekohl) [inline manifests/config.pp:38]: You can consider `bool2str()`:
```suggestion
  ini_setting { "${config_file} register-unsafely-without-email true":
    ensure  => bool2str($unsafe_registration, 'present', 'absent'),
```

File: manifests/config.pp [hunk 2]
```diff
@@ -29,17 +29,23 @@
...
+    path    => $config_file,
+    section => '',
+    setting => 'register-unsafely-without-email',
+    value   => true,
+    require => File[$config_dir],
```
Comment (ekohl) [inline manifests/config.pp:43]: It feels to me that this is something that ini_setting ought to do, so I checked and it's there:
https://github.com/puppetlabs/puppetlabs-inifile/blob/a0ef2bdc68dcb01713f1f3e50a44824e053927a3/lib/puppet/type/ini_setting.rb#L155-L157

It was added in https://github.com/puppetlabs/puppetlabs-inifile/commit/a2b46d20f298b2f3a861a542c2894031766e2829 which is version 2.5.0 and higher. If you bump that in `metadata.json` you can leave this line out.

On the other hand, you are just moving things around.

**Review by ekohl — APPROVED** [review]
Feel free to either address the comments or merge without.

File: manifests/config.pp (hunk 2 above)
Comment (neomilium) [reply manifests/config.pp]: Thanks

**Review by ekohl — APPROVED** [review]

**Pull request merged**